### PR TITLE
75/implement tax lot errors

### LINKED ISSUE #75: Implement tax lot error responses (drizzle)
# Description
From discussion #55, we need to implement error handling for tax lot endpoints. Schema enforcement and database access are primary points of errors. Each of these are accomplished differently in drizzle and mikro. Consequently, they will be implemented separately. This is the ticket to implement drizzle. The code examples are meant to show general structure. Actual implementation may vary.

## Acceptance criteria
- [x] Error handling should be done within the drizzle featureFlag whenever it is an orm-specific handler.
- [x] Custom exceptions implemented in ticket #72 should be used whenever applicable
- [x]  `tax-lots/<bbl>`
  - [x] Throws a 400 when the bbl does not meet the required format [code example](https://github.com/NYCPlanning/ae-zoning-api/blob/5d76bbc0c59a96df36cd831fa2c092aa1e93eae8/src/tax-lot/tax-lot.service.ts#L26)
  - [x] Throws a 404 when no tax lots are returned [code example](https://github.com/NYCPlanning/ae-zoning-api/blob/5d76bbc0c59a96df36cd831fa2c092aa1e93eae8/src/tax-lot/tax-lot.service.ts#L30)
  - [x] Throws a 500 when there is an error accessing the database [code example](https://github.com/NYCPlanning/ae-zoning-api/blob/5d76bbc0c59a96df36cd831fa2c092aa1e93eae8/src/tax-lot/tax-lot.repo.ts#L37)
- [x] `tax-lots/<bbl>/geojson` 
  - [x] Throws a 400 when the bbl does not meet the required format
  - [x] Throws a 404 when no tax lots are returned
  - [x] Throws a 500 when there is an error accessing the database
- [x] `tax-lots/<bbl>/zoning-districts`  
  - [x] Throws a 400 when the bbl does not meet the required format
  - [x] Throws a 404 when no tax lots are returned
  - [x] Throws a 500 when there is an error accessing the database
- [x] `tax-lots/<bbl>/zoning-districts/classes` 
  - [x] Throws a 400 when the bbl does not meet the required format
  - [x] Throws a 404 when no tax lots are returned
  - [x] Throws a 500 when there is an error accessing the database

## Related tickets
blocked by #69 to document the endpoints
blocked by #72 to implement the messages to throw
blocked by #73 to implement zod schemas

### PR DESCRIPTION
**Pull request by TangoYankee**
closes #75 

To stay consistent with the order of documentation added, this ticket should not by merged until after Issues #77

### CODE REVIEW

File: src/tax-lot/tax-lot.service.ts [hunk 1]
```diff
@@ -119,6 +153,13 @@ export class TaxLotService {
...
   async findZoningDistrictClassByTaxLotBbl(bbl: string) {
     if (this.featureFlagConfig.useDrizzle) {
+      let taxLotCheck;
+      try {
+        taxLotCheck = await this.checkTaxLotByBbl().execute({ bbl });
+      } catch {
+        throw DataRetrievalException;
+      }
+      if (taxLotCheck === undefined) throw ResourceNotFoundException;
       const zoningDistrictClasses = await this.db
```
Comment (TangoYankee) [inline src/tax-lot/tax-lot.service.ts:163]: Database access needs to be wrapped here

File: src/tax-lot/tax-lot.service.ts [hunk 2]
```diff
@@ -95,6 +122,13 @@ export class TaxLotService {
...
   async findZoningDistrictByTaxLotBbl(bbl: string) {
     if (this.featureFlagConfig.useDrizzle) {
+      let taxLotCheck;
+      try {
+        taxLotCheck = await this.checkTaxLotByBbl().execute({ bbl });
+      } catch {
+        throw DataRetrievalException;
+      }
+      if (taxLotCheck === undefined) throw ResourceNotFoundException;
       const zoningDistricts = await this.db
```
Comment (TangoYankee) [inline src/tax-lot/tax-lot.service.ts:132]: Database access needs to be wrapped here

File: src/tax-lot/tax-lot.service.ts [hunk 3]
```diff
@@ -53,8 +61,27 @@ export class TaxLotService {
...
     if (this.featureFlagConfig.useDrizzle) {
+      let taxLotCheck;
+      try {
+        taxLotCheck = await this.checkTaxLotByBbl().execute({ bbl });
+      } catch {
+        throw DataRetrievalException;
+      }
+      if (taxLotCheck === undefined) throw ResourceNotFoundException;
+
       const result = await this.db.query.taxLot.findFirst({
```
Comment (TangoYankee) [inline src/tax-lot/tax-lot.service.ts:85]: Database access needs to be wrapped here

File: src/tax-lot/tax-lot.service.ts [hunk 4]
```diff
@@ -28,20 +29,27 @@ export class TaxLotService {
...
-          liFt: false,
-        },
-        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
-        with: {
-          borough: true,
-          landUse: true,
-        },
-      });
-      return result !== undefined ? result : null;
+      let result;
```
Comment (TangoYankee) [inline src/tax-lot/tax-lot.service.ts:32]: Add type defintion

File: src/tax-lot/tax-lot.service.ts [hunk 5]
```diff
@@ -53,8 +61,27 @@ export class TaxLotService {
     }
   }
 
+  checkTaxLotByBbl() {
+    return this.db.query.taxLot
+      .findFirst({
+        columns: {
+          boroughId: true,
```
Comment (TangoYankee) [inline src/tax-lot/tax-lot.service.ts:68]: Replace borughId with bbl

File: src/tax-lot/tax-lot.service.ts [hunk 6]
```diff
@@ -55,61 +74,84 @@ export class TaxLotService {
...
-          block: result.block,
-          lot: result.lot,
-          address: result.address,
-          landUse: result.landUse,
-        },
-        geometry,
-      };
+      let taxLotCheck: Pick<SelectTaxLot, "bbl"> | undefined;
+      try {
+        taxLotCheck = await this.#checkTaxLotByBbl.execute({ bbl });
```
Comment (horatiorosa) [inline src/tax-lot/tax-lot.service.ts:79]: We don't need to do a check before querying the tax lot for this case.

**Pull request merged**